packages/fx-core/src/core/question.ts

```diff
@@ -55,6 +58,33 @@ export const QuestionRootFolder: FolderQuestion = {
   title: "Workspace folder",
 };
 
+export const QuestionSelectTargetEnvironment: SingleSelectQuestion = {
+  type: "singleSelect",
+  name: CoreQuestionNames.TargetEnvName,
+  title: "Select an environment",
+  staticOptions: [],
+  skipSingleOption: true,
+  forgetLastValue: true,
+};
+
+export const QuestionNewTargetEnvironmentName: TextInputQuestion = {
+  type: "text",
+  name: CoreQuestionNames.NewTargetEnvName,
+  title: "New environment name",
+  validation: {
+    validFunc: async (input: string): Promise<string | undefined> => {
+      const targetEnvName = input as string;
+      const match = targetEnvName.match(environmentManager.envNameRegex);
+      if (!match) {
+        return "Environment name can only contain letters, digits, _ and -.";
```

Comment:
If we will use the env name as part of the app name/azure source name, then we may need to make the rule more restrict, like the current app name input validation.

Reply:
Not sure if we have such requirements, as current pattern is `appName` + `suffix` for the resource naming. Maybe we could update it when we have this requirement?

Reply:
I can imagine the app name would be different for different environment (please correct me if I'm wrong), like current teams toolkit, the remote app for the default env use `{app_name}`, while local debug app use `{app_name}_localdebug`.

Maybe this is overthinking, just a remind that if that case happen, we may have to consider backwards compatibility for the old env name with not-allowed characters.

Reply:
Got it, so the question is "will the env name be exposed to user to reference, such like `app name`?". If so, I would prefer to discuss it with PM and take care of it as a standalone task and not cover it in this PR, since the `app name` is a example and there might be other restrictions if the env name is supported be referenced.

Reply:
Since we don't know where the env name will be used, we don't know all the restriction. (user might use it out of our control). So another point is that, it's the consumer's responsibility to format the naming/user input.

Reply:
Sure, let's discuss this with PM first.